write/batcher.go

```diff
@@ -59,13 +66,29 @@ func (b *Batcher) Write(ctx context.Context, org, bucket platform.ID, r io.Reade
 	return nil
 }
 
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
 // read will close the line channel when there is no more data, or an error occurs.
 // it is possible for an io.Reader to block forever; Write's context can be
 // used to cancel, but, it's possible there will be dangling read go routines.
 func (b *Batcher) read(ctx context.Context, r io.Reader, lines chan<- []byte, errC chan<- error) {
 	defer close(lines)
 	scanner := bufio.NewScanner(r)
 	scanner.Split(ScanLines)
+
+	maxLineLength := bufio.MaxScanTokenSize
+	if b.MaxLineLength > 0 {
+		maxLineLength = b.MaxLineLength
+	}
+
+	// limit the initial size of the buffer to a maximum of bufio.MaxScanTokenSize bytes
+	scanner.Buffer(make([]byte, min(bufio.MaxScanTokenSize, maxLineLength)), maxLineLength)
```

Comment:
Or just `scanner.Buffer(nil, maxLineLength)` and let the buffer be grown as needed when lines are read. That way when all lines are short (presumably a common case) we'll use a lot less memory.

Reply:
Good feedback – consider it done!